rig/machine_control/machine_controller.py

```diff
@@ -1479,6 +1529,16 @@ def __len__(self):
         """Return the number of bytes in the file-like view of SDRAM."""
         return self._end_address - self._start_address
 
+    def __enter__(self):
+        """Enter a new block which will call :py:meth:`~.flush` when exited."""
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        """Exit a block and call :py:meth:`~.flush`."""
+        # We don't handle the exception, but we will call flush
+        self.flush()
```

Comment:
To match the common pattern with `with` and files:

```python
with open("/some/file", "r") as f:
    # ...
```

The `__exit__` should probably close the file(?).